webplugin/js/app/km-rich-text-event-handler.js

```diff
@@ -373,36 +373,49 @@ Kommunicate.richMsgEventHandler = {
         var data = {};
         var isActionableForm = (form.className.indexOf("mck-actionable-form") != -1 );
         var replyText = target.title || target.innerHTML;
-        var  inputs = form.getElementsByTagName('input');
-        for(var i = 0; i<inputs.length;i++){
-            if (inputs[i].type == 'radio') {
-                if(inputs[i].checked == true) {
-                    data[inputs[i].name] = inputs[i].value;
-                } 
-            } else if(inputs[i].type == 'checkbox'){
-                if(inputs[i].checked == true) {
-                    !data[inputs[i].name] && (data[inputs[i].name] = []);
-                    data[inputs[i].name].push(inputs[i].value);
-                }
-            } else {
-                data[inputs[i].name] = inputs[i].value;
-                try {
-                    if(inputs[i].dataset.regex) {
-                        validString = Kommunicate.richMsgEventHandler.isValidString(inputs[i].dataset.regex, inputs[i].value);
-                        validationResults.push(validString);
-                        inputElement = form.getElementsByClassName("mck-form-error-"+inputs[i].name.toLowerCase().replace(/ +/g, ""))[0];
-                        if(!validString) {
-                            inputElement.innerHTML = inputs[i].dataset.errorText || MCK_LABELS["rich.form"].errorText;
-                        } else {
-                            inputElement.innerHTML = "";
-                        }
+        var formElements = [];
+        formElements = Array.prototype.concat.apply(formElements, form.getElementsByTagName('input'));
+        formElements = Array.prototype.concat.apply(formElements, form.getElementsByTagName('select'));
+        var name = "";
+        var type = "";
+        var value = "";
+        for(var i = 0; i<formElements.length;i++){
+            name = formElements[i].name;
+            type = formElements[i].type;
+            value = formElements[i].value;
+            switch(type) {
+                case 'radio':
+                    if(formElements[i].checked == true) {
+                        data[name] = value;
+                    } 
+                  break;
+                case 'checkbox':
+                    if(formElements[i].checked == true) {
```

Comment:
@jithinbalu you can add this check if(formElements[i].checked).
If there is a case where this value can be null you can write like this.
 if(formElements[i].checked  != null && formElements[i].checked) {

Reply:
@ab14bhardwaj 
did the changes.